src/Illuminate/Support/HigherOrderCollectionProxy.php

```diff
@@ -54,7 +54,15 @@ public function __get($key)
     public function __call($method, $parameters)
     {
         return $this->collection->{$this->method}(function ($value) use ($method, $parameters) {
-            return $value->{$method}(...$parameters);
+            if (is_object($value)) {
+                return $value->{$method}(...$parameters);
+            }
+
+            if ($method == 'empty') {
```

Comment:
I don't think you need `array_merge` either, the test using `substr(3, -3)` worked as expected